src/api/haventec.common.ts

```diff
@@ -9,13 +10,15 @@ class HaventecCommon {
         return ht_cryptoService.getSaltedPin(pin, salt);
     }
 
-    public getDeviceInfo(detailedFingerprint = false): Object {
-        let deviceInfo = ht_device_infoService.getDeviceInfo(detailedFingerprint);
-        if (deviceInfo) 
-            return { 
-                params: deviceInfo
-            }
-        return;
+    public getDeviceInfo(detailedFingerprint = false): DeviceInfo {
+        let deviceInfoParams = ht_device_infoService.getDeviceInfoParams(detailedFingerprint);
+
+        const deviceInfo: DeviceInfo = new DeviceInfo();
+        if (deviceInfoParams) {
```

Comment:
I see no need to check for `deviceInfoParams` so you mind as well assign it anyway since you are creating a new `deviceInfo` object which will have a default value of null for the params

Reply:
This is a sound check. Its always better to check an object is defined before you use it. I have stated this many times